tck/src/main/java/org/eclipse/microprofile/config/tck/ConvertedNullValueTest.java

```diff
@@ -0,0 +1,90 @@
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information regarding copyright ownership.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * You may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+package org.eclipse.microprofile.config.tck;
+
+import javax.enterprise.context.ApplicationScoped;
+import javax.inject.Inject;
+
+import org.eclipse.microprofile.config.Config;
+import org.eclipse.microprofile.config.inject.ConfigProperty;
+import org.eclipse.microprofile.config.spi.Converter;
+import org.eclipse.microprofile.config.tck.converters.Pizza;
+import org.eclipse.microprofile.config.tck.converters.PizzaConverter;
+import org.jboss.arquillian.container.test.api.Deployment;
+import org.jboss.arquillian.testng.Arquillian;
+import org.jboss.shrinkwrap.api.Archive;
+import org.jboss.shrinkwrap.api.ShrinkWrap;
+import org.jboss.shrinkwrap.api.asset.EmptyAsset;
+import org.jboss.shrinkwrap.api.asset.StringAsset;
+import org.jboss.shrinkwrap.api.spec.JavaArchive;
+import org.jboss.shrinkwrap.api.spec.WebArchive;
+import org.testng.Assert;
+import org.testng.annotations.Test;
+
+/**
+ * This is to test that when a converter returns null for some config value, null will be assigned. This means the
+ * property still exists but the value is set to null on purpose.
+ */
+public class ConvertedNullValueTest extends Arquillian {
+    private @Inject Config config;
+    private @Inject MyBean myBean;
+
+    @Deployment
+    public static Archive deployment() {
+        JavaArchive testJar = ShrinkWrap
+                .create(JavaArchive.class, "ConvertedNullValueTest.jar")
+                .addClasses(ConvertedNullValueTest.class, Pizza.class, PizzaConverter.class, MyBean.class)
+                .addAsManifestResource(
+                        new StringAsset(
+                                "partial.pizza=cheese"),
+                        "microprofile-config.properties")
+                .addAsManifestResource(EmptyAsset.INSTANCE, "beans.xml")
+                .addAsServiceProvider(Converter.class, PizzaConverter.class)
+                .as(JavaArchive.class);
+
+        WebArchive war = ShrinkWrap
+                .create(WebArchive.class, "ConvertedNullValueTest.war")
+                .addAsLibrary(testJar);
+        return war;
+    }
+
+    @Test
+    public void testDefaultValue() {
+        // This should return null as the converter returns null if the property does not contain :
+        Assert.assertNull(myBean.getPizza());
```

Comment:
Hum, shouldn't this fall into the empty rules here: https://github.com/eclipse/microprofile-config/issues/446?

Reply:
And in that case, throw an Exception?

Reply:
not really. The empty rull is to throw exception when a `null` passed to converter. This one is different: `cheese` was passed to converter. Converter decides to return null after parsing the input.

Reply:
Happy to see you confused. This is why I think allowing converters return `null` is no good idea as it is very confusing.

Reply:
In my opinion, empty rules must apply to conversion result as well, or we fall again into the same issue where we are going to see inconsistent results when looking up configuration values.

Reply:
I did say this was an experiment :). In this case, I will do a separate PR for Option 2 to see whether it is any better.